tests/unit/common/middleware/test_container_hierarchy.py

```diff
@@ -0,0 +1,203 @@
+import os.path
+import json
+import sys
+import unittest
+from swift.common import swob, utils
+from swift.common.swob import Request
+from oioswift.common.middleware import container_hierarchy
+
+# Hack PYTHONPATH so "test" is swift's test directory
+sys.path.insert(1, os.path.abspath(os.path.join(__file__, '../../../../..')))
+from test.unit.common.middleware.helpers import FakeSwift  # noqa
+
+
+class OioContainerHierarchy(unittest.TestCase):
+    def setUp(self):
+        conf = {'sds_default_account': 'OPENIO'}
+        self.filter_conf = {
+            'strip_v1': 'true',
+            'swift3_compat': 'true',
+            'account_first': 'true',
+            'create_dir_placeholders': 'true',
+            'recursive_placeholders': 'true',
+            'sds_default_account': 'OPENIO'}
+        self.app = FakeSwift()
+        self.ch = container_hierarchy.filter_factory(
+            conf,
+            **self.filter_conf)(self.app)
+
+    def call_app(self, req, app=None):
+        if app is None:
+            app = self.app
+
+        self.authorized = []
+
+        def authorize(req):
+            self.authorized.append(req)
+
+        if 'swift.authorize' not in req.environ:
+            req.environ['swift.authorize'] = authorize
+
+        req.headers.setdefault("User-Agent", "Melted Cheddar")
+
+        status = [None]
+        headers = [None]
+
+        def start_response(s, h, ei=None):
+            status[0] = s
+            headers[0] = h
+
+        body_iter = app(req.environ, start_response)
+        with utils.closing_if_possible(body_iter):
+            body = b''.join(body_iter)
+
+        return status[0], headers[0], body
+
+    def call_ch(self, req):
+        return self.call_app(req, app=self.ch)
+
+    # check number of request generated by Container Hierarchy
+    def test_simple_put(self):
```

Comment:
I suggest to describe the test in a docstring instead of a comment, to make the description appear in the logs in case of failure.